feat: add additional input parameters and child module for App Insights AVM module

tyconsulting opened the pull request:
## Description

closes #895 

## Pipeline

[![avm.res.insights.component](https://github.com/tyconsulting/bicep-registry-modules/actions/workflows/avm.res.insights.component.yml/badge.svg?branch=users%2Ftao%2F895_appInsights)](https://github.com/tyconsulting/bicep-registry-modules/actions/workflows/avm.res.insights.component.yml)

## Adding a new module

<!--Run through the checklist if your PR adds a new module.-->

- [ ] A proposal has been submitted and approved.
- [ ] I have included "Closes #{module_proposal_issue_number}" in the PR description.
- [ ] I have run `brm validate` locally to verify the module files.
- [ ] I have run deployment tests locally to ensure the module is deployable.

## Updating an existing module

<!--Run through the checklist if your PR updates an existing module.-->

- [ ] This is a bug fix:
  - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
  - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
- [ ] I have run `brm validate` locally to verify the module files.
- [x] I have run deployment tests locally to ensure the module is deployable.
- [x] I have read the [Updating an existing module](https://github.com/Azure/bicep-registry-modules/blob/main/CONTRIBUTING.md#updating-an-existing-module) section in the contributing guide and updated the `version.json` file properly:
  - [ ] The PR contains backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`.
  - [x] The PR contains backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] The PR contains breaking changes, and I have bumped the MAJOR version in `version.json`.
- [x] I have updated the examples in README with the latest module version number.

AlexanderSehr commented:
Hey @tyconsulting, thanks for your contribution. @krbar will review it once he's back in the office :) 
We're on it.

krbar commented on avm/res/insights/component/main.bicep:
```diff
@@ -107,6 +122,14 @@ resource appInsights 'Microsoft.Insights/components@2020-02-02' = {
 }
 
+module linkedStorageAccount 'linkedStorageAccounts/main.bicep' = if (!empty(linkedStorageAccountResourceId)) {
+  name: '${name}-linkedStorageAccount'
```
```suggestion
  name: '${uniqueString(deployment().name, location)}-appInsights-linkedStorageAccount'
```

krbar reviewed (commented):
Hello @tyconsulting, nice contribution, thank you! Looks good to me, I just added a small suggestion to align the deployment name of the child module to the other. A re-generation of the ARM file will be needed.

tyconsulting commented:
> Hello @tyconsulting, nice contribution, thank you! Looks good to me, I just added a small suggestion to align the deployment name of the child module to the other. A re-generation of the ARM file will be needed. Please also re-run the workflow after the change.

hi @krbar thanks for reviewing this PR. I have updated it as per your suggestion. Please take another look. here's the result of the latest pipeline run:

[![avm.res.insights.component](https://github.com/tyconsulting/bicep-registry-modules/actions/workflows/avm.res.insights.component.yml/badge.svg?branch=users%2Ftao%2F895_appInsights)](https://github.com/tyconsulting/bicep-registry-modules/actions/workflows/avm.res.insights.component.yml)

krbar commented:
@Azure/avm-core-team-technical-bicep PR is fine from my point of view and can be merged, please approve.
